.github/workflows/test-and-publish-nightly.yml

```diff
@@ -28,64 +25,53 @@ jobs:
         java-version: [11, 17, 21, 24]
         java-distribution: [corretto]
+        architecture: [x64]
         container: ["public.ecr.aws/async-profiler/asprof-builder-x86:latest"]
         include:
-          - runson:
-              display: macos-14-arm64
-              name: macos-14
+          - name: macos-13
             java-version: 11
             java-distribution: corretto
+            architecture: x64
+            container: ""  # Not using container for mac-os as we have images only for linux
+          - name: macos-14
+            java-version: 21
+            java-distribution: corretto
+            architecture: arm64
             container: ""  # Not using container for mac-os as we have images only for linux
-          - runson:
-              display: linux-arm64
-              name: ubuntu-24.04-arm # There is no "latest" tag available (yet) as ARM runners are still in public preview
+          - name: ubuntu-24.04-arm # There is no "latest" tag available (yet) as ARM runners are still in public preview
             java-version: 11
             java-distribution: corretto
+            architecture: arm64
             container: "public.ecr.aws/async-profiler/asprof-builder-arm:latest"
-    runs-on: ${{ matrix.runson.name }}
+    runs-on: ${{ matrix.name }}
     container:
       image: ${{ matrix.container }}
       options: --privileged
-    name: "test (${{ matrix.runson.display }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }})"
+    name: "test (${{ matrix.name }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }} ${{ matrix.architecture }}"
     steps:
-      - name: Setup Java for macOS x64
-        uses: actions/setup-java@v4
-        if: ${{ matrix.runson.name == 'macos-14' }}
-        with:
-          distribution: ${{ matrix.java-distribution }}
-          java-version: ${{ matrix.java-version }}
-          architecture: x64 # set up for x64, as the default arm one will override this later
-      - name: Setup Java for default architecture
+      - name: Setup Java
         uses: actions/setup-java@v4
         with:
           distribution: ${{ matrix.java-distribution }}
           java-version: ${{ matrix.java-version }}
-          # architecture: not specifying this defaults to architecture of the runner
+          architecture: ${{ matrix.architecture }}
       - name: Checkout sources
         uses: actions/checkout@v4
       - name: Build and test
         id: build
         run: |
           HASH=$(echo ${{ github.sha }} | cut -c-7)
-          case "${{ matrix.runson.name }}" in
-            macos*)
-              make COMMIT_TAG=$HASH FAT_BINARY=true release test -j
-            ;;
-            *)
-              make COMMIT_TAG=$HASH CC=/usr/local/musl/bin/musl-gcc release test -j
-            ;;
-          esac
+          PACKAGE_NAME_SUFFIX="jdk${{ matrix.java-version }}"
```

Comment:
What is it for and why is it needed?

Reply:
See [here](https://github.com/async-profiler/async-profiler/actions/runs/14266840749)

![image](https://github.com/user-attachments/assets/229d3a0b-fdb2-486e-8b3e-2084584cf832)

Due to how we select the OS name in `Makefile`, we don't have a way to discriminate between artifacts produced on `macos-13` or `macos-14`. Thus, without this suffix, GitHub actions complains because two artifacts with the same name are being published (e.g. [here](https://github.com/async-profiler/async-profiler/actions/runs/14266362853)).

This happens to work because we use two different JDKs for `macos-13` and `macos-14`, but it would be probably cleaner to allow injecting OS info to the `Makefile` via a variable.

Reply:
OK, I understand about architecture suffix, but why having jdk suffix? Ideally, contents of a package produced with different JDKs should be identical. Previously we had only one artifact per platform, and that was enough.

Reply:
If each job publishes an artifact, we need to differentiate the name somehow. If the only difference between two jobs is the JDK version, we need to have it in the artifact name. 

Another solution is not to publish from all jobs, maybe we should keep the variable `RELEASE_FROM_JAVA_VERSION`?

Reply:
Right, we don't need to publish from all jobs. My point on `RELEASE_FROM_JAVA_VERSION` was to get rid of the variable tied to a Java version in favor of explicit condition that specifies jobs to publish from.